go-controller/pkg/egressFirewall/egressfirewall.go

```diff
@@ -110,50 +118,45 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 		return errList
 	}
 
-	err = ef.addLogicalRouterPolicyToClusterRouter(nsInfo.addressSet.GetIPv4HashName(), nsInfo.addressSet.GetIPv6HashName(), egressFirewallStartPriorityInt)
+	err = ef.addLogicalRouterPolicyToClusterRouter(address_set.GetIPv4ASHashedName(egressFirewall.Namespace), address_set.GetIPv6ASHashedName(egressFirewall.Namespace), egressFirewallStartPriorityInt)
```

Comment:
This seems a bit confusing; this is implying that namespaces are guaranteed to have an address set based on their specific name.

That's fine, but that's not what these function names imply. It would be better for the structure of the code to reflect this. In other words. Something like `AddressSetsForNS() (string, string)` ?

Reply:
Or, `AddressSetsFor(kind, name)`, where `kind` is an enum of Namespace, DNS domain name, etc.

Reply:
the functions `GetIPv4ASName()`, `GetIPv6ASName()`, `GetIPv4ASHashedName()`, and `GetIPv6ASHashedName()` are used to generate all the address_set names regardless of type. The address_set creation code does not distinguish between the type of object creating it, Are you suggesting to start distinguishing between the objects that create address_sets?

Reply:
You are generating an address set based on the namespace name. However, you don't **qualify** that name as a namespace. Later, if you define a different kind of address set, and also name it by namespace, you will collide.

Reply:
See my thoughts down below... I agree with half of what Casey says, disagree with half :)